test/airport.spec.js

```diff
@@ -0,0 +1,331 @@
+airport.landPlane(plane2); //land plane two
+airport.landPlane(plane3); //try to land plane three but it shouldn't be succesful as airport only has capacity for 2
+expected = 2; // we only expect two planes to land
+actual = airport.getLandedPlaneCount(); //let's see how many actually landed
+
+// Assert - testing the actual results against expected results
+result = assertEquals(expected, actual);
+console.log(`Test 6 - 'landPlane cannot happen': ${result}`);
+//console.log(actual);
+
+// Clean up - here we are clearing the variables ready for next test, to avoid contamination
+airport = null;
+plane = null;
+expected = undefined;
+actual = undefined;
+result = undefined;
+
+console.log(`============================`);
+
+// Test 7 - landPlane of Plane(id) and check if isPlanePresent (T/F)
+//7 landPlane Plane(id) and check if isPlanePresent (T/F)
+//8 in the same way - takeOffPlane Plane(id) and check if isPlanePresent(T / F)
+    
+console.log(`============================`);
+console.log(`Test 7 - landPlane into airport`);
+
+// Arrange - this is setting up the variables required for the test
+airport = new Airport(1); //I'm putting a value in here for defaultAirportCapacity, otherwise 0 is default and no plane can land
+plane1 = new Plane('Delta');
+expected = true;
+
+// Act - this is running the methods to be tested and collecting the actual results
+airport.landPlane(plane1);
+actual = airport.isPlanePresent(plane1);
+
+//Assert - testing the actual results against expected results
+result = assertEquals(expected, actual);
+console.log(`Test 7 - 'Plane is present': ${result}`);
+console.log(`============================`);
+
+// Clean up - here we are clearing the variables ready for next test, to avoid contamination
+airport = null;
+plane = null;
+expected = undefined;
+actual = undefined;
+result = undefined;
+
+//Test 8 - takeOffPlane Plane(id) and check if isPlanePresent(T / F)
+
+console.log(`============================`);
+console.log(`Test 8 - takeOffPlane Plane(id) from airport`);
+
+// Arrange - this is setting up the variables required for the test
+airport = new Airport(1);
+plane1 = new Plane('Delta');
+expected = false;
+
+// Act - this is running the methods to be tested and collecting the actual results
+airport.landPlane(plane1);
+airport.takeOffPlane(plane1);
+actual = airport.isPlanePresent(plane1);
+
+//Assert - testing the actual results against expected results
+result = assertEquals(expected, actual);
+console.log(`Test 8 - 'Plane is not present': ${result}`);
+
+// // Clean up - here we are clearing the variables ready for next test, to avoid contamination
+airport = null;
+plane = null;
+expected = undefined;
+actual = undefined;
+result = undefined;
+
+console.log(`============================`);
+
+//Test 9 - try to take off a plane(id) which is not in the airport (landedPlanes)
+
+console.log(`============================`);
+console.log(`Test 9 - try to takeOffPlane Plane(id) that is not in landedPlanes`);
+
```

Comment:
Wend round in circles and revisited test 9 a few times. Was making it more complicated than it needed to be. Went with a Boolean return in takeOffPlane method in the end, but would rather have a 'throw error'.